Create IoT Firmware Defense SDK

TheDonLuke opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

Commit:
adding generated sdk

Commit:
adding generated sample code

Commit:
adding generated API file

Commit:
regenerating against correct swagger readme

ArcturusZhang commented on sdk/fist/Azure.ResourceManager.IoTFirmwareDefense/README.md:
```diff
@@ -0,0 +1,80 @@
+# Microsoft Azure IoTFirmwareDefense management client library for .NET
+
+**[Describe the service briefly first.]**
```
Could we include some introduction of your RP here?

eatkimchee replied:
Hi @ArcturusZhang the readme has been updated, and we added tests as well. Can you review again please? thanks!

Commit:
adding description to readme

Commit:
adding workspace create and get tests

Commit:
adding workspace delete test

Commit:
renaming workspace test source file.

Since the delete doesn't live in the collection class, I thought it'd be
more appropriate to name the test file just WorkspaceTest

Commit:
updated the recordings when running tests

Commit:
adding firmware tests and recordings

Commit:
adding target framework to sdk csproj

Commit:
removing unneeded proj file

Commit:
fixing targetframework prop to support multiple frameworks.

m-nash commented:
There is no ci run for mgmt iot we definitely need to fix that before this goes out since there is no apiview to review here.

m-nash reviewed (commented):
The capitalism for `IoT` should be `Iot` when used in a compound word.

https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-namespace-naming

m-nash commented:
/azp prepare pipelines

m-nash commented:
/azp run prepare-pipelines

m-nash commented:
/azp run net - iotfirmwaredefense - ci

Commit:
Moving FIST SDK to live under IoT SDK per request.

Commit:
adding nonwords used in the proj to the cspell config

Commit:
adding firmwaredefense yml to excludes as well

eatkimchee commented:
/azp run prepare-pipelines

m-nash commented:
/azp run prepare-pipelines

m-nash commented:
/azp run net - iot - mgmt - ci

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.IotFirmwareDefense](https://apiview.dev/Assemblies/Review/3a9374b65a5b41a58cf6c16f8435ed97)

m-nash commented:
Left some comments in apiview https://apiview.dev/Assemblies/Review/3a9374b65a5b41a58cf6c16f8435ed97/4594a9acd08046e7a3153c0742e1cd5e

Commit:
renaming items per request

Commit:
updating generated samples for new names

Commit:
adding cves to cspell file

eatkimchee commented:
@m-nash   we have addressed things from the apiview review and left some responses for you. Can you check it again with the latest build?

m-nash commented:
> @m-nash we have addressed things from the apiview review and left some responses for you. Can you check it again with the latest build?

Left a few replies and I have one follow up item on the `bool` vs extensible enum I will get back to you on.  Nothing in my opinion is a blocker for beta with the couple additional prefix type names that I commented on.  Everything else can be covered in GA timeframe.

eatkimchee committed:
rename Workspace to FirmwareWorkspace as per apiview feedback

eatkimchee committed:
rename Cve to FirmwareCve as per apiview feedback

eatkimchee committed:
updated tests to support resource class rename

eatkimchee commented:
@m-nash added the prefixes you asked for

eatkimchee commented:
@ArcturusZhang  Hi can you please approve this PR and merge it if we met all the requirements? thanks!

ArcturusZhang reviewed (commented):
Generally looks good to me, but I did not see the api file in your PR, could you please run the script `{Root}/eng/Export-API.ps1 iot` to generate it and push it to your branch?

eatkimchee committed:
Ran Export-API.ps1 to generate the api file

eatkimchee committed:
Merge branch 'Azure:main' into create-velma-sdk

eatkimchee committed:
Regenerated code, samples, and api after rebasing
